go/filemanager/pfsmod/ls.go

```diff
@@ -0,0 +1,203 @@
+package pfsmod
+
+import (
+	"errors"
+	"flag"
+	log "github.com/golang/glog"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+const (
+	lsCmdName = "ls"
+)
+
+//LsResult is LsCmd's result
+type LsResult struct {
+	Path    string `json:"Path"`
+	ModTime string `json:"ModTime"`
+	Size    int64  `json:"Size"`
+	IsDir   bool   `json:"IsDir"`
+}
+
+//LsCmd is LsCommand structure
+type LsCmd struct {
+	Method string
+	R      bool
+	Args   []string
+}
+
+//ToURLParam encoding LsCmd to URL Encoding string
+func (p *LsCmd) ToURLParam() string {
+	parameters := url.Values{}
+	parameters.Add("method", p.Method)
+	parameters.Add("r", strconv.FormatBool(p.R))
+
+	for _, arg := range p.Args {
+		parameters.Add("arg", arg)
+	}
+
+	return parameters.Encode()
+
+}
+
+//ToJSON does't need to be implemented
+func (p *LsCmd) ToJSON() ([]byte, error) {
+	return nil, nil
+}
+
+//NewLsCmdFromFlag returen a new LsCmd
+func NewLsCmdFromFlag(f *flag.FlagSet) (*LsCmd, error) {
+	cmd := LsCmd{}
+
+	cmd.Method = lsCmdName
+	cmd.Args = make([]string, 0, f.NArg())
+
+	var err error
+	f.Visit(func(flag *flag.Flag) {
+		if flag.Name == "r" {
+			cmd.R, err = strconv.ParseBool(flag.Value.String())
+			if err != nil {
+				panic(err)
+			}
+		}
+	})
+
+	for _, arg := range f.Args() {
+		log.V(2).Info(arg)
+		cmd.Args = append(cmd.Args, arg)
+	}
+
+	return &cmd, nil
+}
+
+//NewLsCmdFromURLParam returns a new LsCmd according path variable
+func NewLsCmdFromURLParam(path string) (*LsCmd, error) {
+	cmd := LsCmd{}
+
+	m, err := url.ParseQuery(path)
+	if err != nil ||
```

Comment:
`if err != nil` just indicate the parsing failed, it's not "not enough arguments", maybe we should use a different StatusText for it?